client/consensus/slots/src/lib.rs

```diff
@@ -201,9 +201,10 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		// deadline our production to 98% of the total time left for proposing. As we deadline
 		// the proposing below to the same total time left, the 2% margin should be enough for
 		// the result to be returned.
+		let inherent_data = slot_info.create_inherent_data.create_inherent_data().await.ok()?;
```

Comment:
This should be done before calculating `proposing_remaining_duration`

And an error should be logged.